tasks/tasks/causemos_processors_test.py

```diff
@@ -0,0 +1,415 @@
+
+    rd = iteration_func(index_model_object)
+    datasets_ids = download_datasets(rd)
+
+    out = build_synthetic_dataset(datasets_ids, rd)
+
+    normalized = normalize_synthetic_dataset(out)
+
+    pca_details = perform_pca(normalized)
+
+    weights = pca_to_weights_v3(pca_details, target_pca_explained_sum=0.9)
+
+    rounded_as_percent = list(map(lambda i: f"{roundToFour(i) * 100}%", weights.tolist()))
+
+    assert rounded_as_percent == ['4.73%', '5.7700000000000005%', '5.62%', '5.8999999999999995%', '4.07%', '5.66%', '0.0%', '4.65%', '5.47%', '5.140000000000001%', '5.0%', '3.53%', '4.4799999999999995%', '4.47%', '4.68%', '4.91%', '5.029999999999999%', '4.8500000000000005%', '4.9799999999999995%', '5.45%', '5.609999999999999%']
+
+
+@pytest.mark.integration
+def test_pca_to_weights_v2_control_data():
+
+    pca_details = (
+        numpy.array(
+            [[1.13900827e-03, 4.01488796e-02, 7.06536188e-01,
+              7.06536188e-01, 0.00000000e+00],
+             [7.13923827e-01, 6.99690962e-01, 1.93044911e-02,
+              1.93044911e-02, 0.00000000e+00],
+             [7.00222444e-01, 7.13316638e-01, 2.08315041e-02,
+              2.08315041e-02, 0.00000000e+00],
+             [2.11047685e-17, 2.19081849e-17, 7.07106781e-01,
+              7.07106781e-01, 0.00000000e+00],
+             [0.00000000e+00, 0.00000000e+00, 0.00000000e+00,
+              0.00000000e+00, 1.00000000e+00]]
+        ),
+
+        5,
+
+        numpy.array(
+            [5.00405570e-01, 2.60246340e-01, 2.39348091e-01,
+             2.51310581e-32, 0.00000000e+00]
+        )
+    )
+
+    weights = pca_to_weights_v2(pca_details)
+
+    rounded_as_percent = list(map(lambda i: f"{roundToFour(i) * 100}%", weights))
+
+    assert rounded_as_percent == ['32.45%', '33.32%', '17.11%', '17.11%', '0.0%']
+
+
+
+@pytest.mark.integration
+def test_pca_to_weights_v2_control_data():
+
+    pca_details = (
+        numpy.array(
+            [[1.13900827e-03, 4.01488796e-02, 7.06536188e-01,
+              7.06536188e-01, 0.00000000e+00],
+             [7.13923827e-01, 6.99690962e-01, 1.93044911e-02,
+              1.93044911e-02, 0.00000000e+00],
+             [7.00222444e-01, 7.13316638e-01, 2.08315041e-02,
+              2.08315041e-02, 0.00000000e+00],
+             [2.11047685e-17, 2.19081849e-17, 7.07106781e-01,
+              7.07106781e-01, 0.00000000e+00],
+             [0.00000000e+00, 0.00000000e+00, 0.00000000e+00,
+              0.00000000e+00, 1.00000000e+00]]
+        ),
+
+        5,
+
+        numpy.array(
+            [5.00405570e-01, 2.60246340e-01, 2.39348091e-01,
+             2.51310581e-32, 0.00000000e+00]
+        )
+    )
+
+    weights = pca_to_weights_v2(pca_details)
+
+    rounded_as_percent = list(map(lambda i: f"{roundToFour(i) * 100}%", weights))
+
+    assert rounded_as_percent == ['32.45%', '33.32%', '17.11%', '17.11%', '0.0%']
+
+
+
+@pytest.mark.integration
+def test_pca_to_weights_v2_real_index_data():
```

Comment:
@brandomr this is the v2 "updated matrix" (hah) algorithm I had worked before, running against the real ndgain dataset output after performing pca on it.